### PR TITLE
CLN/DEPR: remove Block._holder, deprecated Block.is_categorical

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -227,7 +218,13 @@ def _can_hold_na(self) -> bool:
     @final
     @property
     def is_categorical(self) -> bool:
-        return self._holder is Categorical
+        warnings.warn(
```
Comment (jreback) [inline pandas/core/internals/blocks.py:221]: is there something that actually tests this warning?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/blocks.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:221]: no, its a DeprecationWarning, just for eg pyarrow

**Review by jreback — APPROVED** [review]
lgtm ping on green

**Comment by jbrockmendel** [comment]
ping

**Comment by jorisvandenbossche** [comment]
> no, its a DeprecationWarning, just for eg pyarrow

Small correction: for fastparquet

**Pull request merged**